ee/server/service/mdm.go

```diff
@@ -288,9 +288,6 @@ func (svc *Service) validateMDMAppleSetupPayload(ctx context.Context, payload fl
 	if err != nil {
 		return err
 	}
-	if !ac.MDM.EnabledAndConfigured {
-		return fleet.ErrMDMNotConfigured
-	}
```

Comment:
This contains the `end_user_authentication` config which is currently under macos_setup (see discussion [here](https://fleetdm.slack.com/archives/C03C41L5YEL/p1761244662828529)). Since any platform can now use that setting, we can't gate this behind Apple MDM being turned on.

We should definitely consider moving the `end_user_authentication` setting (and perhaps others) out of `macos_setup`, and replacing this gate but only for the `macos_setup` object?